### DISCUSSION TITLE
Any NextJS 9 with MobX 7 ssr demo?

### DISCUSSION THREAD

**Discussion by kmvan** [vercel/next.js#17793] [Help]
The tutorial demo is outdate and not works with mobx 7.

**Comment by alaminut** [comment]
I've been trying to get mobx 7 working with next 9 as well but tutorial is outdated and it doesn't work. I get server / client content did not match errors and "component X cannot be updated while Y being rendered" errors.

**Reply by kmvan** [reply]
With hooks?

**Reply by alaminut** [reply]
Yes with hooks and mobx-react-lite here is a sample repo https://github.com/alaminut/next-mobx-bug-reproduction

**Reply by kmvan** [reply]
I think that is the point: `import { Provider } from "mobx-react";` 
You may need use `mobx-react-lite` and `observer()` wrapper function.

**Reply by alaminut** [reply]
That doesn't make a difference. I was using mobx-react-lite and React's context before switching to mobx-react's provider to test it. You should probably see that version in the commit history.
